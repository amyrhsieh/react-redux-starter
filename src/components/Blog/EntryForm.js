import React from 'react';
import { render } from 'react-dom';
import { Form, Field } from 'react-final-form';

import styles from './entryForm.scss';

const onSubmit = values => {
  window.alert(JSON.stringify(values, 0, 2));
}

const EntryForm = () => (
  <Form
    onSubmit={onSubmit}
    render={({ handleSubmit, form, submitting, pristine, values }) => (
      <form onSubmit={handleSubmit}>
        <div>
          <label className="formLabel">New Entry</label>
          <Field
            name="content"
            component="textarea"
            placeholder="Write your blog entry here"
          />
        </div>
        <div className="buttons">
          <button type="submit" disabled={submitting || pristine}>
            Submit
          </button>
          <button
            type="button"
            onClick={form.reset}
            disabled={submitting || pristine}
          >
            Reset
          </button>
        </div>
        <pre>{JSON.stringify(values, 0, 2)}</pre>
      </form>
    )}
  />
);

export default EntryForm;
