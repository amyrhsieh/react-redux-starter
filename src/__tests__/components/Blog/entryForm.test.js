import React from 'react';
import EntryForm from 'Components/Blog/EntryForm';
import { shallow } from 'enzyme';

const onSubmit = () => {return true;}

describe('EntryForm (Snapshot)', () => {
  it('EntryForm component renders EntryForm', () => {
    const wrapper = shallow(<EntryForm onSubmit={onSubmit} />);
    expect(wrapper).toMatchSnapshot();
  });
});
