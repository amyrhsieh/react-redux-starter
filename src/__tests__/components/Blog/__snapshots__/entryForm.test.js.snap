// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryForm (Snapshot) EntryForm component renders EntryForm 1`] = `
<ReactFinalForm
  onSubmit={[Function]}
  render={[Function]}
/>
`;
